sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -570,12 +594,12 @@ public Mono<MsalToken> authenticateWithUsernamePassword(TokenRequestContext requ
      */
     @SuppressWarnings("deprecation")
     public Mono<MsalToken> authenticateWithPublicClientCache(TokenRequestContext request, IAccount account) {
-        return publicClientApplicationAccessor.getValue()
+        return getPublicClientInstance(request).getValue()
             .flatMap(pc -> Mono.fromFuture(() -> {
                 SilentParameters.SilentParametersBuilder parametersBuilder = SilentParameters.builder(
                     new HashSet<>(request.getScopes()));
 
-                if (request.getClaims() != null) {
+                if (request.isCaeEnabled() && request.getClaims() != null) {
```

Comment:
Now we will only add the claims if CAE is enabled and there are claims? Are there no cases where there was previously a claim that we need to honor that is not a CAE situation?

Reply:
previously, it was enabled by default for Public client auth.
With this change, it is disabled by default, and needs to be enabled via the TRC parameter.
This will be a behavioral breaking change, being done by design across languages.
We should update TSG with this info.

Reply:
Will you include the TSG update here or in another change?

Reply:
Yeah, I'll add that in a follow up PR to TSG, as that is not shipped.